Sources/MessagingPush/Integration/CioAppDelegateWithoutTokenRetrieval.swift

```diff
@@ -0,0 +1,171 @@
+import CioInternalCommon
+import UIKit
+
+public typealias CioAppDelegateType = NSObject & UIApplicationDelegate
+
+public typealias ConfigInstance = () -> MessagingPushConfigOptions
+
+public typealias UserNotificationCenterInstance = () -> UserNotificationCenterIntegration
+
+// sourcery: AutoMockable
+public protocol UserNotificationCenterIntegration {
+    var delegate: UNUserNotificationCenterDelegate? { get set }
+}
+
+extension UNUserNotificationCenter: UserNotificationCenterIntegration {}
+
+private extension UIApplication {
+    func cioRegisterForRemoteNotifications(logger: Logger) {
+        logger.debug("CIO: Registering for remote notifications")
+        registerForRemoteNotifications()
+    }
+}
+
+@available(iOSApplicationExtension, unavailable)
+open class CioAppDelegateWithoutTokenRetrieval: CioAppDelegateType, UNUserNotificationCenterDelegate {
+    @_spi(Internal) public let messagingPush: MessagingPushInstance
+    @_spi(Internal) public let logger: Logger
+    @_spi(Internal) public var implementedOptionalMethods: Set<Selector> = [
+        // UIApplicationDelegate
+        #selector(UIApplicationDelegate.application(_:didFinishLaunchingWithOptions:)),
+        #selector(UIApplicationDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:)),
+        #selector(UIApplicationDelegate.application(_:didFailToRegisterForRemoteNotificationsWithError:)),
+        #selector(UIApplicationDelegate.application(_:continue:restorationHandler:)),
+        // UNUserNotificationCenterDelegate
+        #selector(UNUserNotificationCenterDelegate.userNotificationCenter(_:willPresent:withCompletionHandler:)),
+        #selector(UNUserNotificationCenterDelegate.userNotificationCenter(_:didReceive:withCompletionHandler:)),
+        #selector(UNUserNotificationCenterDelegate.userNotificationCenter(_:openSettingsFor:))
+    ]
+
+    @_spi(Internal) public var config: ConfigInstance?
+
+    private var userNotificationCenter: UserNotificationCenterInstance?
+    private let wrappedAppDelegate: UIApplicationDelegate?
+    private var wrappedNoticeCenterDelegate: UNUserNotificationCenterDelegate?
```

Comment:
nit: the naming `notice` is not easily recognizable and mismatches the actual name